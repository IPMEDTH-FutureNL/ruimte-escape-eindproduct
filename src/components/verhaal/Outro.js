import React, { useState } from 'react';

import "../../css/outro.css"
import Astronaut from "../../img/gameplayImages/characters/character2_standard_schuin_mouthopen.svg"
import AstronautTwee from "../../img/gameplayImages/characters/character4_standard_schuin_mouthclosed.svg"
import AstronautDrie from "../../img/gameplayImages/characters/character1_standard_mouthclosed.svg"
import AstronautVier from "../../img/gameplayImages/characters/character3_standard_mouthopen.svg"

import { FaAngleRight } from "react-icons/fa";
import TextCloud from '../../img/gameplayImages/snippets/textcloud.png'

import PopUp from '../PopUp';

const Outro = ({pause}) => {
    const [text, setText] = useState("Welkom op Mars " + localStorage.getItem("playerOne") +  " en " + localStorage.getItem("playerTwo") + " we zijn veilig aangekomen dankzij jullie hulp!");
    const [count, setCount] = useState(0);
    

    const story = (newText) => {
        setCount(count + newText);

        switch(count){
            case 1:
                setText("Jullie kunnen jullie zelf nu echte ruimtevaarders noemen!")
                break;
            case 2:
                let openpopup = document.getElementById("popup");
                openpopup.style.display = "block";
                let textCloud = document.getElementById("textcloud");
                textCloud.style.display = "none";
                break;
        }
    }

    return (
        <section className="container__outro">
                <section className="container__gameplay-succes" id="container__gameplay">
                    <div className="container__astronaut">
                        <div className="astronaut">
                            <img src={Astronaut} alt="astronaut" className="astronaut__image"></img>
                        </div>
                        <div className="textcloud" id="textcloud">
                            <img src={TextCloud} alt="tekst balloon" className="textcloud__image"></img>
                            <div className="textcloud__field" id="textcloud__field">
                                <p className="textcloud__text">{text}</p>
                                <FaAngleRight className="continue" onClick={() => {story(1); pause();}}/>
                            </div>
                        </div>
                        <div className="astronaut__twee">
                        <div className="astronaut">
                            <img src={AstronautTwee} alt="astronaut" className="astronaut__image"></img>
                        </div>
                        </div>
                        <div className="astronaut__drie">
                            <div className="astronaut">
                                <img src={AstronautDrie} alt="astronaut" className="astronaut__image"></img>
                            </div>
                        </div>
                        <div className="astronaut__vier">
                            <div className="astronaut">
                                <img src={AstronautVier} alt="astronaut" className="astronaut__image"></img>
                            </div>  
                        </div>
                    </div>
                </section>

                <PopUp text="Jullie hebben de escaperoom met succes gespeeld! Bedankt voor jullie inzet!" navigate="true"></PopUp>
        </section>
    )

}

export default Outro;