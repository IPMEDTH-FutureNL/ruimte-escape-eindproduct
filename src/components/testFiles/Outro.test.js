import renderer from 'react-test-renderer';
import Outro from "../verhaal/Outro";
import {BrowserRouter as Router} from 'react-router-dom';

it('Test if Outro component is rendered', () => {
    const component = renderer.create(
        <Router>
            <Outro></Outro>,
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})