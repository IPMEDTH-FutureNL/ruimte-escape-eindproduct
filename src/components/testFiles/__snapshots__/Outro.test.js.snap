// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Outro component is rendered 1`] = `
Array [
  <section
    className="container__outro"
  >
    <section
      className="container__gameplay-succes"
      id="container__gameplay"
    >
      <div
        className="container__astronaut"
      >
        <div
          className="astronaut"
        >
          <img
            alt="astronaut"
            className="astronaut__image"
            src="character2_standard_schuin_mouthopen.svg"
          />
        </div>
        <div
          className="textcloud"
          id="textcloud"
        >
          <img
            alt="tekst balloon"
            className="textcloud__image"
            src="textcloud.png"
          />
          <div
            className="textcloud__field"
            id="textcloud__field"
          >
            <p
              className="textcloud__text"
            >
              Welkom op Mars null en null we zijn veilig aangekomen dankzij jullie hulp!
            </p>
            <svg
              className="continue"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 256 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </div>
        </div>
        <div
          className="astronaut__twee"
        >
          <div
            className="astronaut"
          >
            <img
              alt="astronaut"
              className="astronaut__image"
              src="character4_standard_schuin_mouthclosed.svg"
            />
          </div>
        </div>
        <div
          className="astronaut__drie"
        >
          <div
            className="astronaut"
          >
            <img
              alt="astronaut"
              className="astronaut__image"
              src="character1_standard_mouthclosed.svg"
            />
          </div>
        </div>
        <div
          className="astronaut__vier"
        >
          <div
            className="astronaut"
          >
            <img
              alt="astronaut"
              className="astronaut__image"
              src="character3_standard_mouthopen.svg"
            />
          </div>
        </div>
      </div>
    </section>
    <div
      className="pop-up"
      id="popup"
    >
      <p
        className="pop-up-text"
      >
         
        Jullie hebben de escaperoom met succes gespeeld! Bedankt voor jullie inzet!
         
      </p>
      <button
        className="button primary"
        onClick={[Function]}
      >
        Verder!
      </button>
    </div>
  </section>,
  ",",
]
`;
